app/models/claim.rb

```diff
@@ -140,6 +141,22 @@ def payroll_gender_missing?
     %w[male female].exclude?(payroll_gender)
   end
 
+  def awaiting_check?
+    submitted? && check.nil?
+  end
+
+  def approaching_deadline?
```

Comment:
I can't help but wonder, what if we had a `days_until_checking_deadline` that returns either a positive or negative number and used that to determine where the claim is on the schedule and also to generate warning markup? Would that make things simpler?

Reply:
Yeah it would make it a lot simpler. It would mean that I can remove three methods from the claim model and simplify the `deadline_warning` helper at the same time